src/feelpp/benchmarking/reframe/report.py

```diff
@@ -41,50 +33,91 @@ class Report:
 
 
     def processData(self):
+        # Avoids error for reporting scale performances not launched with 'CpuVariation.py'
+        try:
+            self.toolbox = self.data['runs'][0]['testcases'][0]['check_vars']['toolbox']
+        except KeyError as e:
+            self.toolbox = ''
+
         # Needed dataframe for building df_perf and df_speedup
         df = pd.DataFrame(self.data['runs'][0]['testcases'][0:])
         df['num_tasks'] = df['check_vars'].apply(lambda x: x['num_tasks'])
         df['num_tasks'] = df['num_tasks'].astype(int)
         df = df[['num_tasks', 'perfvars']]
 
+        if self.partial:
+            self.buildPartialDict()
         self.buildPerf(df)
         self.buildSpeedup(df)
+        self.linearReg(self.df_speedup)
+        if self.partial:
+            self.linearReg(self.df_partialSpeedup)
+
+
+    def buildPartialDict(self):
+        tmpDict = {}
+        for perfvar in self.data['runs'][0]['testcases'][0]['perfvars']:
+            name = perfvar['name']
+            stageName = name.split('_')[0]
 
+            if stageName not in tmpDict.keys():
+                tmpDict[stageName] = []
+            tmpDict[stageName].append(name)
+
+        self.partialDict = {}
+        for stageName, perfNameLst in tmpDict.items():
+            perfNameLst.pop()
+            perfNameLst = [name.split('_')[1] for name in perfNameLst]
+            self.partialDict.update( {stageName : perfNameLst} )
+
+
+    def isPartialPerf(self, perfName):
+        if self.partial:
+            for nameLst in self.partialDict.values():
+                for name in nameLst:
+                    if name in perfName:
+                        return True
+            return False
+        else:
+            return False
 
 
     def buildPerf(self, df):
         for k, t in enumerate(df['num_tasks'].unique()):
             for perf in df['perfvars'][k]:
-                    if isPartialPerf(perf['name']):
-                        self.df_partialPerf = pd.concat([self.df_partialPerf, pd.DataFrame(
-                            [{'num_tasks': t, 'name': perf['name'], 'value': perf['value']}])], ignore_index=True)
-                    else:
-                        self.df_perf = pd.concat([self.df_perf, pd.DataFrame(
-                            [{'num_tasks': t, 'name': perf['name'], 'value': perf['value']}])], ignore_index=True)
+                name = perf['name']
+                nameWithoutPrefix = name.split('_',1)[-1]
+
+                if self.isPartialPerf(nameWithoutPrefix):
+                    self.df_partialPerf = pd.concat([self.df_partialPerf, pd.DataFrame(
+                        [{'num_tasks': t, 'name': name, 'value': perf['value']}])], ignore_index=True)
+                else:
+                    self.df_perf = pd.concat([self.df_perf, pd.DataFrame(
+                        [{'num_tasks': t, 'name': name, 'value': perf['value']}])], ignore_index=True)
 
         self.df_perf['name'] = self.df_perf['name'].astype(str)
```

Comment:
Use the `dtype` parameter of `pd.DataFrame` to optimise.

Reply:
from pandas doc:
http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.html
```
dtype, default None
    Data type to force. Only a single dtype is allowed. If None, infer.
```